scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala

```diff
@@ -197,8 +197,29 @@ class BigQueryClient private (private val projectId: String,
       getTable(table).getSchema
     }
 
-  /** Execute a query and save results into a temporary table. */
-  def queryIntoTable(sqlQuery: String): QueryJob = {
+  /** Make a query and save results to a destination table. */
+  def queryIntoTable(sqlQuery: String,
+                     destinationTable: TableReference,
+                     flattenResults: Boolean): Job = {
+    val queryJob = makeQueryJob(sqlQuery, destinationTable, flattenResults)
+    queryJob.waitForResult()
+    val jobRef = queryJob.jobReference.get
+    bigquery.jobs().get(projectId, jobRef.getJobId).execute()
+  }
+
+  /** Make a query and save results to a destination table. */
+  def queryIntoTable(sqlQuery: String,
+                     destinationTable: String,
+                     flattenResults: Boolean = false): Job =
+    queryIntoTable(sqlQuery, BigQueryIO.parseTableSpec(destinationTable), flattenResults)
+
+  /**
+   * Create a new query job and save results into a temporary table.
+   *
+   * No query will be executed if a cached table exists. Query execution is delayed until
+   * `waitForResult` is called on the result `QueryJob`.
+   */
+  def newQueryJob(sqlQuery: String, flattenResults: Boolean): QueryJob = {
```

Comment:
This should probably not be user facing. Make it private and keep the name?
No need for default if not user facing.

Also maybe I should create a user facing version `def query` that blocks and return `Job` instead?